akka-stream-tests/src/test/scala/akka/stream/scaladsl/SubStreamMemoryLeaks.scala

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.scaladsl
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+
+import scala.io.StdIn
+
+object SubStreamMemoryLeaks extends App {
+
+  implicit val system = ActorSystem()
+  implicit val mat = ActorMaterializer()
+
+  StdIn.readLine()
+
+  Source.fromIterator(() ⇒
+    Iterator.continually(Source.single(Array.fill(1024 * 1024)(1)))
+  ) // each incoming of these will cause creation of a new async callback
+    // which would leak because of bug 24046
+    .flatMapConcat(identity)
+    .grouped(100)
+    .runWith(Sink.foreach(_ ⇒ print(".")))
+
+}
```

Comment:
I don't mind keeping it around if you want, but do you intentionally want to keep it?

Reply:
I couldn't figure out a reasonable way to repeat the leak with a test, so this is the only repeater there is. I guess if it's just a standalone app here it won't be of much use. I'll remove it.